src/main/scala/com/fulcrumgenomics/util/Rscript.scala

```diff
@@ -0,0 +1,91 @@
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+package com.fulcrumgenomics.util
+
+import java.nio.file.Path
+
+import dagr.commons.util.LazyLogging
+
+import scala.io.Source
+import scala.util.{Success, Try}
+
+/**
+  * Object that enables running of R scripts via the Rscript command line too.
+  */
+object Rscript extends LazyLogging {
+  /** The name of the Rscript executable. */
+  private val Executable = "Rscript"
+
+  /** Exception class that holds onto an executable's exit/status code. */
+  case class RscriptException(status: Int) extends RuntimeException {
+    override def getMessage: String = s"Rscript failed with exit code ${status}."
+  }
+
+  /** Returns true if R is available and false otherwise. */
+  lazy val Available: Boolean = {
+    try {
+      val process = new ProcessBuilder(Executable, "-e", "require(ggplot2)").redirectErrorStream(true).start()
+      process.waitFor() == 0
+    }
+    catch { case e: Exception => false }
+  }
+
+  /** Executes an Rscript from the classpath if Rscript is available. */
+  def execIfAvailable(scriptResource: String, args: String*): Try[Unit] =
+    if (Available) exec(scriptResource, args:_*) else Success[Unit]()
+
+  /** Executes an Rscript from a script stored at a Path if Rscript is available. */
+  def execIfAvailable(script: Path, args: String*): Try[Unit] =
+    if (Available) exec(script, args:_*) else Success[Unit]()
+
+  /** Executes an Rscript from the classpath. */
+  def exec(scriptResource: String, args: String*): Try[Unit] =
+    Try { writeResourceToTempFile(scriptResource) }.map(path => exec(path, args:_*))
+
+  /** Executes an Rscript from a script stored at a Path. */
+  def exec(script: Path, args: String*): Try[Unit] = Try {
+    val command = Rscript.Executable +: script.toAbsolutePath.toString +: args
+    val process = new ProcessBuilder(command:_*).redirectErrorStream(true).start()
+    val pipe    = Io.pipeStream(process.getInputStream, logger.info)
+    val retval  = process.waitFor()
+    pipe.close()
+
+    if (retval != 0) throw RscriptException(retval)
+  }
+
+  /** Extracts a resource from the classpath and writes it to a temp file on disk. */
+  private def writeResourceToTempFile(resource: String): Path = {
+    val in = getClass.getClassLoader.getResourceAsStream(resource)
+    if (in == null) throw new IllegalArgumentException(s"Resource '${resource} not found in classpath")
+    val path = Io.makeTempFile("script.", ".R")
+    path.toFile.deleteOnExit()
+    val out = Io.toWriter(path)
+
+    Source.fromInputStream(in).getLines().foreach { l => out.write(l); out.newLine() }
+    in.close()
+    out.close()
+    path
+  }
+}
```

Comment:
I agree that would be good to do, but it's also a bunch of work and might make our builds less robust.  I'm not sure what to do about that.